src/ModelContextProtocol/Client/McpClient.cs

```diff
@@ -25,43 +30,61 @@ internal sealed class McpClient : McpEndpoint, IMcpClient
     /// <param name="options">Options for the client, defining protocol version and capabilities.</param>
     /// <param name="serverConfig">The server configuration.</param>
     /// <param name="loggerFactory">The logger factory.</param>
-    public McpClient(IClientTransport clientTransport, McpClientOptions options, McpServerConfig serverConfig, ILoggerFactory? loggerFactory)
+    public McpClient(IClientTransport clientTransport, McpClientOptions? options, McpServerConfig serverConfig, ILoggerFactory? loggerFactory)
         : base(loggerFactory)
     {
         _clientTransport = clientTransport;
+
+        if (options?.ClientInfo is null)
+        {
+            options = options?.Clone() ?? new();
```

Comment:
I had to do a similar thing for ServerCapabilities in McpServer as part of #115. I did basically the same thing, but instead of adding a Clone() method to ServerOptions, I manually cloned ServerCapabilities into a new McpServer property to avoid shallow copying the entire options type.

There's nothing wrong with the options shallow copy since we're still not mutating the passed-in options, but I prefer cloning just the capabilities type if we can. But mostly, I'd prefer consistency. So, if we think shallow copying the options type and using that is the source of truth is better, we should update McpServer to do the same. I do like having the Clone() method in the type being shallow copied. So, I definitely be supportive of adding ServerCapabilities.Clone().

Reply:
I ended up refactoring things to not need the clone.